private registries support

### The problem

When using private registry with authorization required (e.g. Nexus or verdaccio), it's impossible to use patch-package, when using [per-project auth](https://docs.npmjs.com/files/npmrc#per-project-config-file)

The reason is that patch-package tries to install package in temp folder, but since there's no `.npmrc` file with auth token - 401 error is thrown:

```bash
npm ERR! code E401
```

Note, that when using private registries, all the packages from NPM registry are usually proxied. Installing any of them requires user to be authorized.

### Potential solution

I've managed to make it work locally by copying `.npmrc` file to the tmp repo.
My suggestion is to check if `.npmrc` file exists and copy it to temp repo. Also, maybe it's worth to consider copying `.yarnrc` as well

I can submit a PR :)

--- a/src/makePatch.ts
+++ b/src/makePatch.ts
@@ -102,6 +102,12 @@ export function makePatch({
       "package.json",
     )).version as string
 
+    // copy .npmrc in case if packages are hosted in private registry
+    const npmrcPath = join(appPath, ".npmrc")
+    if (existsSync(npmrcPath)) {
+      copySync(npmrcPath, join(tmpRepo.name, ".npmrc"))
+    }
+
     if (packageManager === "yarn") {
       console.info(
         chalk.grey("•"),

